cmd/kubernetes.go

```diff
@@ -0,0 +1,818 @@
+			die("could not open resource file with path: %s", err)
+		}
+		decoder := gob.NewDecoder(file)
+		err = decoder.Decode(resources)
+		if err != nil {
+			die("error decoding resource file: %s", err)
+		}
+
+		// now check if the ssh forwarding is up
+		fmPidFile := filepath.Join(config.ManagerDir, "kubernetes_resources.fw.pid")
+		fmPid, fmRunning := checkProcess(fmPidFile)
+
+		// try and stop the remote manager
+		noManagerMsg := "; deploy first or use --force option"
+		noManagerForcedMsg := "; tearing down anyway - you may lose changes if not backing up the database to S3!"
+		serverHadProblems := false
+		if fmRunning {
+			jq := connect(1*time.Second, true)
+			if jq != nil {
+				var syncMsg string
+				if internal.IsRemote(config.ManagerDbBkFile) {
+					if _, errf := os.Stat(config.ManagerDbFile); !os.IsNotExist(errf) {
+						// move aside the local database so that if the manager is
+						// started locally, the database will be restored from S3
+						// and have the history of what was run in the cloud
+						if errf = os.Rename(config.ManagerDbFile, config.ManagerDbFile+".old"); err == nil {
+							syncMsg = "; the local database will be updated from S3 if manager started locally"
+						} else {
+							warn("could not rename the local database; if the manager is started locally, it will not be updated with the latest changes in S3! %s", errf)
+						}
+					}
+				} else {
+					// copy the remote database locally, so if the manager is
+					// started locally we have the history of what was run in
+					// the cloud. The gap between backing up and shutting down
+					// is "fine"; though some db writes may occur, the user
+					// obviously doesn't care about them. On recovery we won't
+					// break any pipelines.
+					errf := jq.BackupDB(config.ManagerDbFile)
+					if errf != nil {
+						msg := "there was an error trying to sync the remote database: " + errf.Error()
+						if forceTearDown {
+							warn(msg + noManagerForcedMsg)
+						} else {
+							die(msg)
+						}
+					}
+					syncMsg = " and local database updated"
+				}
+
+				ok := jq.ShutdownServer()
+				if ok {
+					info("the remote wr manager was shut down" + syncMsg)
+				} else {
+					msg := "there was an error trying to shut down the remote wr manager"
+					if forceTearDown {
+						warn(msg + noManagerForcedMsg)
+						serverHadProblems = true
+					} else {
+						die(msg)
+					}
+				}
+			} else {
+				msg := "the remote wr manager could not be connected to in order to shut it down"
+				if forceTearDown {
+					warn(msg + noManagerForcedMsg)
+					serverHadProblems = true
+				} else {
+					die(msg + noManagerMsg)
+				}
+			}
+		} else {
+			if forceTearDown {
+				warn("the deploy port forwarding is not running, so the remote manager could not be stopped" + noManagerForcedMsg)
+				serverHadProblems = true
+			} else {
+				die("the deploy port forwarding is not running, so can't safely teardown" + noManagerMsg)
+			}
+		}
```

Comment:
How similar is the above block to that's in the teardown for openstack scheduler? Seems like both cmds should call a function that does all this...

Reply:
I think it could be, I'll take a look and see how I might do it